libraries/provider_hab_service.rb

```diff
@@ -0,0 +1,76 @@
+# Copyright:: Copyright 2016, Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/provider/service"
+
+class Chef
+  class Provider
+    class Service
+      class HabSup < Chef::Provider::Service
+        include HabitatCookbook::Helpers
+
+        use_inline_resources
+
+        provides :hab_service
+
+        action :start do
+          declare_resource(:systemd_unit, "#{service_name}.service") do
```

Comment:
i don't think that has been backported into compat_resource so that won't fix anything here.

having said that, i think we aggressively drop backcompat in this cookbook.  if it winds up being `chef >= 12.16.38` so be it.